easypermissions/src/main/java/pub/devrel/easypermissions/EasyPermissions.java

```diff
@@ -399,82 +226,35 @@ private static void notifyAlreadyHasPermissions(Object object,
      * rationale, false otherwise.
      */
     private static boolean shouldShowRationale(@NonNull Object object, @NonNull String[] perms) {
-        boolean shouldShowRationale = false;
-        for (String perm : perms) {
-            shouldShowRationale =
-                    shouldShowRationale || shouldShowRequestPermissionRationale(object, perm);
-        }
-        return shouldShowRationale;
-    }
-
-    /**
-     * @param activity Activity
-     * @param perms Array of permissions
-     * @return true if the user has previously denied any of the {@code perms} and we should show a
-     * rationale, false otherwise.
-     */
-    public static boolean somePermissionDenied(@NonNull Activity activity, @NonNull String[] perms) {
-        return shouldShowRationale(activity, perms);
-    }
-
-    /**
-     * @param fragment Fragment
-     * @param perms Array of permissions
-     * @return true if the user has previously denied any of the {@code perms} and we should show a
-     * rationale, false otherwise.
-     */
-    public static boolean somePermissionDenied(@NonNull Fragment fragment, @NonNull String[] perms) {
-        return shouldShowRationale(fragment, perms);
+        return PermissionHelper.getInstance(object).shouldShowRationale(perms);
     }
 
     /**
-     * @param fragment Fragment
+     * @param host {@link Object} requesting permissions.
      * @param perms Array of permissions
      * @return true if the user has previously denied any of the {@code perms} and we should show a
      * rationale, false otherwise.
      */
-    @RequiresApi(api = Build.VERSION_CODES.M)
-    public static boolean somePermissionDenied(@NonNull android.app.Fragment fragment, @NonNull String[] perms) {
-        return shouldShowRationale(fragment, perms);
+    public static boolean somePermissionDenied(@NonNull Object host, @NonNull String[] perms) {
```

Comment:
While we're refactoring, I just noticed: why aren't we using `vargs`? It would be a small yet sweet improvement.